File: Java/Math/Squares Difference.java
//https://www.geeksforgeeks.org/problems/squares-difference0939/1
class Solution {
    static long squaresDiff(int N) {
        long sumOfSquares = (long) N * (N + 1) * (2L * N + 1) / 6;
        long squareOfSum = (long) (N * (N + 1) / 2);
        squareOfSum *= squareOfSum;
        
        return Math.abs(sumOfSquares - squareOfSum);
    }
}

File: Python/Math/Squares Difference.py
//https://www.geeksforgeeks.org/problems/squares-difference0939/1
class Solution:
    def squaresDiff (self, N):
        # code here 
        return abs((N * (N + 1) * (2 * N + 1)) // 6 - pow(N * (N + 1) // 2, 2))
